workshop/7 - Performance Efficiency/README.md

```diff
@@ -0,0 +1,216 @@
+# 7- Performance Efficiency
+
+Performance efficiency is the ability of a workload to scale and meet the demands placed on it by users in an efficient manner. In cloud environments, a workload should anticipate increases in demand to meet business requirements.
+
+## The Cache-Aside pattern
+
+The Cache-Aside pattern is a technique that's used to manage in-memory data caching. It reduces the request response time and can lead to increased response throughput. This efficiency reduces the number of horizontal scaling events, making the app more capable of handling traffic bursts. It also improves service availability by reducing the load on the primary data store and decreasing the likelihood of service outages.
+
+### Implementing Cache-Aside in Relecloud Lite
+
+### Cache-Aside in Relecloud Concerts
+Take a look for the Cache-Aside pattern implementation in our main application.
+
+1. Open the **Relecloud.sln** solution.
+1. Note that we use **Microsoft.Extensions.Caching.StackExchangeRedis** for Cacheing.
+1. Open the **Startup.cs** of the **Relecloud.Web.CallCenter.Api** project and browse to the `AddDistributedSession` method.
+1. Look at the Azure Redis Cache connection string from the configuration settings.
+
+    ```csharp
+    var redisCacheConnectionString = Configuration["App:RedisCache:ConnectionString"];
+    ```
+1. This code uses Azure Redis if the connection string contains a value
```

Comment:
this should be updated to make use of a managed identity based connection string. the initialization is a bit different - take a look at the updated sample in main